syntax = "proto3";

import "google/protobuf/descriptor.proto";

package p4;

message P4_int {
   string switch_id = 1;
   int64 version = 2;
   int64 replication = 3;
   int64 c = 4;
   int64 hop_count_exceeded = 5;
   int64 inst_count = 6;
   int64 Max_hop_count = 7;
   repeated P4_int_metadata p4_int_metadata = 8;
}

message P4_int_metadata {
   string switch_id = 1;
   int64 c_plane_state_ver_number = 2;
   Ingress ingress = 3;
   Egress egress = 4; 
   Buffer buffer = 5; 
}

message Ingress {
    int64 port_id = 1;
    string timestamp = 2;
    int64 port_RX_pkt_count = 3;
    int64 port_RX_byte_count = 4;
    int64 port_RX_drop_count = 5;
    int64 port_RX_utilization = 6;
}
message Egress {
   int64 port_id = 1; 
   int64 port_RX_pkt_count = 2;
   int64 port_RX_byte_count = 3; 
   int64 port_RX_drop_count = 4; 
   int64 port_RX_utilization = 5; 
}
message Buffer {
  int64 queue_id = 1; 
  int64 inst_queue_len = 2; 
  int64 avg_queue_len = 3; 
  int64 congestion_status = 4; 
  int64 queue_drop_count = 5; 
}
